glean_parser/templates/rust_server.jinja2

```diff
@@ -0,0 +1,308 @@
+    client_info: ClientInfo,
+    ping_info: PingInfo,
+    metrics: Metrics,
+    events: Vec<GleanEvent>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct LogEnvelope {
+    timestamp: String,
+    logger: String,
+    #[serde(rename = "type")]
+    log_type: String,
+    fields: Ping,
+}
+
+impl GleanEventsLogger {
+    fn create_client_info(&self) -> ClientInfo {
+        // Fields with default values are required in the Glean schema, but not used in server context
+        ClientInfo {
+            telemetry_sdk_build: "glean_parser v15.0.2.dev17+g81fec69a".to_string(),
+            fist_run_date: "Unknown".to_string(),
+            os: "Unknown".to_string(),
+            os_version: "Unknown".to_string(),
+            architecture: "Unknown".to_string(),
+            app_build: "Unknown".to_string(),
+            app_display_version: self.app_display_version.clone(),
+            app_channel: self.app_channel.clone(),
+        }
+    }
+
+    fn create_ping_info() -> PingInfo {
+        // times are ISO-8601 strings, e.g. "2023-12-19T22:09:17.440Z"
+        let now = Utc::now().to_rfc3339();
+        PingInfo {
+            seq: 0,
+            start_time: now.clone(),
+            end_time: now,
+        }
+    }
+
+    fn create_ping(
+        &self,
+        document_type: &str,
+        config: &RequestInfo,
+        payload: &PingPayload,
+    ) -> Ping {
+        let payload_json =
+            serde_json::to_string(payload).expect("unable to marshal payload to json.");
+        let document_id = Uuid::new_v4().to_string();
+        Ping {
+            document_namespace: self.app_id.clone(),
+            document_type: document_type.to_string(),
+            document_version: "1".to_string(),
+            document_id,
+            user_agent: Some(config.user_agent.clone()),
+            ip_address: Some(config.ip_address.clone()),
+            payload: payload_json,
+        }
+    }
+
+    /// Method called by each ping-specific record method.
+    /// The goal is to construct the ping, wrap it in the envelope
+    /// and print to stdout.
+    fn record(
+        &self,
+        document_type: &str,
+        request_info: &RequestInfo,
+        metrics: Metrics,
+        events: Vec<GleanEvent>,
+    ) {
+        let telemetry_payload: PingPayload = PingPayload {
+            client_info: self.create_client_info(),
+            ping_info: GleanEventsLogger::create_ping_info(),
+            metrics,
+            events,
+        };
+
+        let ping: Ping = self.create_ping(document_type, request_info, &telemetry_payload);
+
+        let envelope: LogEnvelope = LogEnvelope {
+            timestamp: Utc::now().timestamp().to_string(),
+            logger: "glean".to_string(),
+            log_type: GLEAN_EVENT_MOZLOG_TYPE.to_string(),
+            fields: ping,
+        };
+        let envelope_json =
+            serde_json::to_string(&envelope).expect("unable to marshal payload to json.");
+        println!("{}", envelope_json);
```

Comment:
We probably shouldn't just print to stdout?
Not sure what the exact plan was here, but wouldn't we want to integrate with `log`? Or somehow pass a logger in?

Reply:
I think in the discussions with @akkomar and @travis79 , we had decided stdout is the way to go for rust server metrics, however I could be mistaken. I will double check with them, but I think this is how it's expected to be picked up.

Reply:
How do you normally output logs in Sync @taddes? It would be best to use your standard logging mechanism. We just need to make sure that this log goes to Cloud Logging in a correct format.

Reply:
I think it's good to go with the approach recommended by SRE/SecOps in the template. I don't remember why Go template looks this way, maybe that's what they do in their code.